pkg/daemon/ceph/osd/volume.go

```diff
@@ -179,12 +179,12 @@ func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceO
 			return fmt.Errorf("failed to unmarshal ceph-volume report json. %+v", err)
 		}
 
-		if path.Join("/dev", a.metadataDevice) != cvReport.Vg.Devices {
+		if path.Join("/dev", md) != cvReport.Vg.Devices {
 			return fmt.Errorf("ceph-volume did not use the expected metadataDevice [%s]", a.metadataDevice)
```

Comment:
should the error be changed by `md` too?

Reply:
Yes it should.  I will update that as well.